### PR TITLE
Removed the use of RestartPlan class in the Kafka Roller Class

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change

_Select the type of your PR_

- Refactoring

### Description
This is the starting step for making the process of KafkaRoller class more clean and effective. This PR removes the use `RestartPlan` class. The `restartPlan` method was earlier making use of the `RestartPlan` instance for the on describing how the restart needs to be performed. Instead of creating seperate objects for them we now shifted the parameters of `RestartPlan` to `RestartContext itself to more or less use a single object ie `RestartContext` instance for determining the restart plan. 

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java [hunk 1]
```diff
@@ -342,15 +324,15 @@ private void restartIfNecessary(PodRef podRef, RestartContext restartContext)
         }
 
         try {
-            RestartPlan restartPlan = restartPlan(podRef, pod, restartContext);
-            if (restartPlan.forceRestart || restartPlan.needsRestart || restartPlan.needsReconfig) {
-                if (!restartPlan.forceRestart && deferController(podRef, restartContext)) {
+            restartPlan(podRef, pod, restartContext);
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java:327]: Myabe you should give the method some better name if it doesn't produce restart plan anymore?

**Review by tombentley — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java (hunk 1 above)
Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java:326]: Yeah, that would work for me.

Comment (ShubhamRwt) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java]: Done

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
@ShubhamRwt The regression failed twice on the same test -> can you have a look if it is related or not?

**Comment by ShubhamRwt** [comment]
Fixed the regression test

**Pull request merged**